Frameworks/Foundation/NSURLConnection.mm

```diff
@@ -109,8 +115,13 @@ @implementation NSURLConnection {
 /**
  @Status Interoperable
 */
-+ (BOOL)canHandleRequest:(id)request {
-    return ([NSURLProtocol _URLProtocolClassForRequest:request] != nil) ? YES : NO;
++ (BOOL)canHandleRequest:(NSURLRequest*)request {
+    Class protocolClass = [NSURLProtocol _URLProtocolClassForRequest:request];
+    if (!protocolClass) {
+        return NO;
+    }
+
+    return [protocolClass canInitWithRequest:request];
```

Comment:
you can keep this folded, actually. `[nil canInitWithRequest:request]` returns NO